src/Symfony/Component/Lock/Store/FlockStore.php

```diff
@@ -72,10 +72,17 @@ private function lock(Key $key, bool $blocking)
             return;
         }
 
+        $sanitizedKey = preg_replace('/[^a-z0-9\._-]+/i', '-', $key);
+        if (\strlen($sanitizedKey) > 64) {
+            $sanitizedKey = substr($sanitizedKey, 0, 50);
+            $hashEntropy = 13;
+        } else {
+            $hashEntropy = 7;
+        }
         $fileName = sprintf('%s/sf.%s.%s.lock',
             $this->lockPath,
-            preg_replace('/[^a-z0-9\._-]+/i', '-', $key),
-            strtr(substr(base64_encode(hash('sha256', $key, true)), 0, 7), '/', '_')
+            $sanitizedKey,
```

Comment:
this could be the only line changed in this file: `substr($sanitizedKey, 0, 50),`
I don't think the rest of the logic makes any significant difference. Eg 7 for entropy should be good for all situations (or raised all the time)

Reply:
done